src/compas/geometry/predicates/predicates_2.py

```diff
@@ -392,6 +395,118 @@ def is_polygon_in_polygon_xy(polygon1, polygon2):
         return False
 
 
+def polygon_to_polygon_relationship_xy(polygon1, polygon2):
+    """Determine if a polygon (polygon1) is inside, intersects, is outside or contains another polygon (polygon2) on the XY-plane.
+
+    Parameters
+    ----------
+    polygon1 : sequence[point] | :class:`~compas.geometry.Polygon`
+        List of XY(Z) coordinates of points representing the locations of the corners of the polygon to examine the position regarding the boundary polygon (Z will be ignored).
+        The vertices are assumed to be in order. The polygon is assumed to be closed:
+        the first and last vertex in the sequence should not be the same.
+    polygon2 : sequence[point] | :class:`~compas.geometry.Polygon`
+        List of XY(Z) coordinates of points representing the locations of the corners of the boundary polygon (Z will be ignored).
+        The vertices are assumed to be in order. The polygon is assumed to be closed:
+        the first and last vertex in the sequence should not be the same.
+
+    Returns
+    -------
+    int
+        1 if polygon1 is inside polygon2.
+        0 if polygon1 intersects polygon2.
+        -1 if polygon1 is outside polygon2.
+        -2 if polygon1 contains polygon2 (polygon2 is inside polygon1).
+    """
+    for i in range(len(polygon1)):
+        line1 = [polygon1[i], polygon1[i - 1]]
+        for j in range(len(polygon2)):
+            line2 = [polygon2[j], polygon2[j - 1]]
+            if is_intersection_segment_segment_xy(line1, line2):
+                return 0
+    for pt in polygon1:
+        if is_point_in_polygon_xy(pt, polygon2):
+            return 1
+    if is_point_in_polygon_xy(polygon2.centroid, polygon1):
+        return -2
```

Comment:
why the different approach here?

Reply:
You mean different because of calling other existing methods inside the loop here ?